__tests__/integration.test.js

```diff
@@ -102,37 +108,64 @@ beforeEach(() => {
           expect(article).toHaveProperty("votes");
           expect(article).toHaveProperty("article_img_url");
           expect(article).toHaveProperty("comment_count");
-      })  
-    })
-
-  })
-    test("should return article objects which have the correct data type values", () => {
-      return request(app)
-      .get("/api/articles")
-      .expect(200)
-      .then((response) => {
-        const articles = response.body.articles;
-        articles.forEach((article) => {
           expect(article).toMatchObject({
             author: expect.any(String),
             title: expect.any(String),
             article_id: expect.any(Number),
             topic: expect.any(String),
             created_at: expect.any(String),
             article_img_url: expect.any(String),
-            comment_count: expect.any(String)
-        })
-      })
-      
-  })
-})
+            comment_count: expect.any(String),
+          });
+        });
+      });
+  });
   test("articles should be ordered by date descending", () => {
     return request(app)
-    .get("/api/articles")
-    .expect(200)
-    .then((response) => {
-      const articles = response.body.articles;
-    expect(articles).toBeSortedBy("created_at", {descending: true})
-  })
-})
-})
\ No newline at end of file
+      .get("/api/articles")
+      .expect(200)
+      .then((response) => {
+        const articles = response.body.articles;
+        expect(articles).toBeSortedBy("created_at", { descending: true });
+      });
+  });
+});
+
+describe("/api/articles/:article_id/comments", () => {
+  test("should return a status 200 and an array of comments for specified article id in descending order", () => {
+    return request(app)
+      .get("/api/articles/3/comments")
+      .expect(200)
+      .then((response) => {
+        const comments = response.body.comments;
+        expect(comments.length).toBe(2);
+        expect(comments).toBeSortedBy("created_at", { descending: true });
+        comments.forEach((comment) => {
+          expect(comment).toMatchObject({
+            body: expect.any(String),
+            votes: expect.any(Number),
+            author: expect.any(String),
+            comment_id: expect.any(Number),
+            article_id: expect.any(Number),
+            created_at: expect.any(String),
+          });
+        });
+      });
+  });
+  test("should return a 404 - not found if the article id doesn't exist", () => {
+    return request(app)
+      .get("/api/articles/35433/comments")
+      .expect(404)
+      .then((response) => {
+        expect(response.body.msg).toBe("404: Article does not exist");
+      });
+  });
+  test("should return a 400 status if the request made is not in the expected form", () => {
+    return request(app)
+      .get("/api/articles/socks_with_flip_flops/comments")
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe("400: Bad request");
+      });
```

Comment:
Nice sad path. 

Please add a test for an article with no comments. You should receive an empty array not an error :)